beamer/state_machine.py

```diff
@@ -121,21 +121,27 @@ def _find_request_by_request_hash(
     return None
 
 
-def _invalidation_ready_for_l1_relay(claim: Claim) -> bool:
+def _invalidation_ready_for_l1_relay(
+    claim: Claim, target_chain_id: ChainId, request_manager: Contract
+) -> bool:
+    target_chain_finalization = request_manager.functions.finalizationTimes(target_chain_id).call()
```

Comment:
We don't necessarily need an event for this, we can just query it on startup and store it somewhere

Reply:
> We don't necessarily need an event for this, we can just query it on startup and store it somewhere

Those were mine thoughts too, but then I remembered that `setFinalizationTime` can be called multiple times.
Which is maybe not what we'd like to allow -- in that case, yes, I'd say look it up on startup and we're done.

Otherwise, we need to consider the case where the finalization time has been changed. (Not sure whether we really need that.)